### PR TITLE
Feature/handle vote transaction

### PR DESCRIPTION
**Pull request by zhangheng1536**
**What does this PR do?**
Deal with voting transactions.
**Why are these changes required?**
Make statistics on the votes.
**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: src/main/java/org/tron/core/db/Manager.java [hunk 1]
```diff
@@ -145,6 +150,42 @@ public boolean processTrx(TransactionCapsule trxCap) {
     return true;
   }
 
+  private void voteWitessCount(Transaction trx) {
+    try {
+      int parameterSize = trx.getParameterList().size();
+      if (parameterSize <= 0) {
```
Comment (KamilZielinski) [inline src/main/java/org/tron/core/db/Manager.java]: You can use `isEmpty()` method to check if list has nothing inside

File: src/main/java/org/tron/core/db/Manager.java [hunk 2]
```diff
@@ -145,6 +150,42 @@ public boolean processTrx(TransactionCapsule trxCap) {
     return true;
   }
 
+  private void voteWitessCount(Transaction trx) {
```
Comment (KamilZielinski) [inline src/main/java/org/tron/core/db/Manager.java]: `voteWitessCount` -> voteWitnessCount

File: src/main/java/org/tron/core/db/Manager.java [hunk 3]
```diff
@@ -145,6 +150,42 @@ public boolean processTrx(TransactionCapsule trxCap) {
...
+            countVoteWitess(voteAddress, voteAdd);
+          });
+        }
+      }
+    } catch (InvalidProtocolBufferException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void countVoteWitess(ByteString voteAddress, int countAdd) {
```
Comment (KamilZielinski) [inline src/main/java/org/tron/core/db/Manager.java]: `countVoteWitess` -> `countVoteWitness`

Comment (zhangheng1536) [reply src/main/java/org/tron/core/db/Manager.java]: All right, this is really my mistake, thank you for your correction, and I will fix it immediately.
@KamilZielinski

File: src/main/java/org/tron/core/db/Manager.java [hunk 4]
```diff
@@ -145,6 +150,40 @@ public boolean processTrx(TransactionCapsule trxCap) {
     return true;
   }
 
+  private void voteWitnessCount(Transaction trx) {
+    try {
+      if (trx.getParameterList() == null || trx.getParameterList().isEmpty()) {
```
Comment (KamilZielinski) [inline src/main/java/org/tron/core/db/Manager.java]: Actually, my bad here. I didn't notice that Apache Commons Collections are included into the project, so you can use `CollectionUtils.isEmpty()` which already has null check

Comment (zhangheng1536) [reply src/main/java/org/tron/core/db/Manager.java]: @KamilZielinski 
We will be happy to accept your PR which  have not  bugs if you wish.
Thank you for your attention.

**Pull request merged**